### PR TITLE
[WiP] Enable and enforce FindBugs in external-monitor-job-plugin

### PR DESCRIPTION
**Pull request by oleg-nenashev**
@reviewbybees

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
FindBugs discovered 7 issue... => WiP

**Comment by oleg-nenashev** [comment]
@reviewbybees Probably the review has not been triggered correctly

File: src/main/java/hudson/model/ExternalJob.java [hunk 1]
```diff
@@ -78,6 +80,7 @@ public synchronized ExternalRun newBuild() throws IOException {
             try {
                 Thread.sleep(1000 - timeSinceLast);
             } catch (InterruptedException e) {
+                //Do nothing
```
Comment (jtnord) [inline src/main/java/hudson/model/ExternalJob.java]: // TODO update to Jenkins ≥ 1.597

File: src/main/java/hudson/model/ExternalRun.java [hunk 2]
```diff
@@ -23,6 +23,7 @@
  */
 package hudson.model;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 import hudson.Proc;
 import hudson.util.DecodingStream;
```
Comment (jtnord) [inline src/main/java/hudson/model/ExternalRun.java]: this is now an unused import.

File: src/main/java/hudson/model/ExternalRun.java [hunk 3]
```diff
@@ -116,7 +117,7 @@ private String elementText(XMLStreamReader r) throws XMLStreamException {
             }
 
             public Result run(BuildListener listener) throws Exception {
-                PrintStream logger = new PrintStream(new DecodingStream(listener.getLogger()));
+                final PrintStream logger = listener.getLogger();
```
Comment (jtnord) [inline src/main/java/hudson/model/ExternalRun.java]: was there not a reason that this was going to log the hex output?

Comment (oleg-nenashev) [inline src/main/java/hudson/model/ExternalRun.java]: Hmm, I've occasionally removed the DecodingStream

**Comment by jtnord** [comment]
@oleg-nenashev comments inline

**Comment by oleg-nenashev** [comment]
@jtnord 
Thanks! The DecodedStream issue must be fixed

File: src/main/java/hudson/model/ExternalJob.java (hunk 1 above)
Comment (jglick) [inline src/main/java/hudson/model/ExternalJob.java]: Yes, good catch, this code is now obsolete.

**Comment by oleg-nenashev** [comment]
marked as WiP

**Comment by recena** [comment]
@oleg-nenashev I think this https://github.com/jenkinsci/external-monitor-job-plugin/pull/11 supersedes to your PR.

**Comment by oleg-nenashev** [comment]
agreed

**Pull request closed without merging**